### ISSUE TITLE
Add support for scanned tasks to various Linux plugins 

### ISSUE THREAD

**Issue by eve-mem** [volatilityfoundation/volatility3#924]
A lot of the linux plugins start from a task struct and provide lots of extra information about the processes. e.g. envvars, lsof, sockstat, psaux. 

It would be great to allow this functionality to task structs found by scanning too. However the question is how to allow actually the user to this.

**Comment by eve-mem** [comment]
I've made no major progress on this yet. I still do think it's a useful feature to add. I'll work on a simple example and see if it works well before adding support for everything.

**Comment by ikelos** [comment]
Removing the stale flag and bumping because this sounds like it might one day be a useful feature to have.  I need to revisit that draft at some point...

**Comment by eve-mem** [comment]
That makes 400 days stale.... But i do think one day this will be useful to do.

**Comment by ikelos** [comment]
Yep, happy to keep the idea here.  We can probably add an "idea" flag and then prevent those going stale, it's just whether we'd ever come back and look at them again?  5:S

**Comment by eve-mem** [comment]
Yah could do. In some ways i appreciate the guilt trip of an idea going stale, reminds me to try and think about it again...

**Comment by ikelos** [comment]
Yeah, it's nice to have the nudge.  Ok, I'll leave it as is then and we can just keep marking it as not stale until one day we forget and it drifts quietly into oblivion...  5;P

**Comment by atcuno** [comment]
Is this still on your roadmap @eve-mem ?

**Comment by eve-mem** [comment]
Hello @atcuno - it is, but so far I haven't found a sample where this ends up be actually useful. 

Most of the things I'm doing end up needing an mm struct to get to the processes private memory, but nearly all the scanned tasks I've ever seen don't have a valid mm anymore. 

I dusted off this bash code I had been using if you wanted to take a look. 
https://github.com/eve-mem/volatility3/blob/linux_bash_scanning/volatility3/framework/plugins/linux/bash.py

I think as a concept it is useful to add this feature, I just haven't had a useful situation come across my desk. If you have an example you could share with me I'd love to work on it.

**Comment by atcuno** [comment]
ok yes, the `mm_struct` will become invalid upon exit. I think for usefulness you would need to examine which `task_struct` members are embedded directly within the structure as base types as these have the highest chance to survive after de-allocation (the ones on the first page will be in tact at least if we are finding the `task_struct` in general). This is the same idea as psscan for Windows, we only touch base type members directly on the `_EPROCESS` for the same reasons.

**Comment by eve-mem** [comment]
That makes an awful lot of sense.

**Comment by eve-mem** [comment]
Probably makes more sense than to scan for the results e.g. bash command lines in the case of the bash plugin rather than work down from the task to find them. 

E.g. similar to sockscan https://github.com/volatilityfoundation/volatility3/pull/1120

For things like envvars I'll have a look into what's embedded and see if i can work out a plan. 

Thank you for that comment, it's really helped me think about the problem.
